src/FoldersView/FolderSourceItem.vala

```diff
@@ -61,41 +61,46 @@ public class Mail.FolderSourceItem : Granite.Widgets.SourceList.ExpandableItem {
             badge = "%d".printf (folderinfo.unread);
         }
 
-        switch (folderinfo.flags & Camel.FOLDER_TYPE_MASK) {
-            case Camel.FolderInfoFlags.TYPE_INBOX:
-                icon = new ThemedIcon ("mail-inbox");
-                can_modify = false;
-                break;
-            case Camel.FolderInfoFlags.TYPE_OUTBOX:
-                icon = new ThemedIcon ("mail-outbox");
-                can_modify = false;
-                break;
-            case Camel.FolderInfoFlags.TYPE_TRASH:
-                icon = new ThemedIcon (folderinfo.total == 0 ? "user-trash" : "user-trash-full");
-                can_modify = false;
-                badge = null;
-                break;
-            case Camel.FolderInfoFlags.TYPE_JUNK:
-                icon = new ThemedIcon ("edit-flag");
-                can_modify = false;
-                break;
-            case Camel.FolderInfoFlags.TYPE_SENT:
-                icon = new ThemedIcon ("mail-sent");
-                can_modify = false;
-                break;
-            case Camel.FolderInfoFlags.TYPE_ARCHIVE:
-                icon = new ThemedIcon ("mail-archive");
-                can_modify = false;
-                badge = null;
-                break;
-            case Camel.FolderInfoFlags.TYPE_DRAFTS:
-                icon = new ThemedIcon ("folder-documents");
-                can_modify = false;
-                break;
-            default:
-                icon = new ThemedIcon ("folder");
-                can_modify = true;
-                break;
+        
+        // (Camel.FolderInfo.Flags & Camel.FOLDER_TYPE_MASK) does not return expected values for multiple types of folders.
+        // Original comparisons are preserved if/when this bug is fixed in libcamel.
+        var masked_flag = folderinfo.flags & Camel.FOLDER_TYPE_MASK;
+        var lowercase_full_name = full_name.down ();
+
+        if (masked_flag == Camel.FolderInfoFlags.TYPE_INBOX ||
+        lowercase_full_name.contains ("inbox")) {
```

Comment:
Formatting nitpick: I'm not sure if there is an elementary guideline for this, but I think these continued `if` conditions should either be indented with 4 or 8 spaces, not 0 relative to the `if` statement.

Reply:
I just found the elementary Code guidelines and i don't believe it does. This is not really a normal case to begin with so I can see why this might be missed.